src/ContaBancaria.java

```diff
@@ -0,0 +1,98 @@
+
+/**
+ * INACABADA
+ */
+public class ContaBancaria {
+	private String agencia;
+	private String conta;
+	private double saldo;
+	private String tipoConta;
+	private boolean ativo;
+	
+	public ContaBancaria() {
+		super();
+	}
+	public ContaBancaria(String agencia, String conta, double saldo, String tipoConta, boolean ativo) {
+		super();
+		this.agencia = agencia;
+		this.conta = conta;
+		this.saldo = saldo;
+		this.tipoConta = tipoConta;
+		this.ativo = ativo;
+	}
+	public String getAgencia() {
+		return agencia;
+	}
+	public void setAgencia(String agencia) {
+		this.agencia = agencia;
+	}
+	public String getConta() {
+		return conta;
+	}
+	public void setConta(String conta) {
+		this.conta = conta;
+	}
+	public double getSaldo() {
+		return saldo;
+	}
+	public void setSaldo(double saldo) {
+		this.saldo = saldo;
+	}
+	public String getTipoConta() {
+		return tipoConta;
+	}
+	public void setTipoConta(String tipoConta) {
+		this.tipoConta = tipoConta;
+	}
+	public boolean isAtivo() {
+		return ativo;
+	}
+	public void setAtivo(boolean ativo) {
+		this.ativo = ativo;
+	}
+	
+	public Transacao sacar(double valor) {
+	    Transacao transacao = new Transacao();
+	    if(valor < this.saldo) {
+	    	transacao.setValor(valor);
+	    } else {
+	    	System.out.println("Valor do saque maior que valor de saldo");
+	    }
+	    return transacao;
```

Comment:
Vejo 3 problemas aqui.
1. O valor só é debitado caso seja menor que o saldo, mas e se quisermos sacar 20 reais de uma conta que tem 20 reais de saldo? não conseguimos.
2. a função só retorna uma transação, mas não atualiza o saldo.
3. Independente se a pessoa tem saldo ou não, estamos retornando uma transação, e nenhuma delas tem o status e a conta que está a criando.
```suggestion
	public Transacao sacar(double valor) {
	    Transacao transacao = new Transacao();
	     transacao.setContaOrigem(this);
	    if(valor <= this.saldo) {
	    	transacao.setValor(valor);
	    	transacao.setStatus("aprovado");
	    	this.saldo -= valor;
	    } else {
	    	System.out.println("Valor do saque maior que valor de saldo");
	    	transacao.setStatus("cancelado");
	    }
	    return transacao;
```